pkg/manager/cluster.go

```diff
@@ -82,65 +81,84 @@ func (c *Cluster) GenerateClusterBundle(bundleDir string) (string, error) {
+
+func wrap(ns string, getter NamespacedGetter) GetRuntimeObjectListFunc {
+	wrapped := func() (runtime.Object, error) {
+		return getter(ns)
+	}
+	return wrapped
 }
 
-func (c *Cluster) generateSupportBundleYAMLsForKubernetes(dir string, errLog io.Writer) {
-	getListAndEncodeToYAML("events", c.sbm.k8s.GetAllEventsList, dir, errLog)
-	getListAndEncodeToYAML("pods", c.sbm.k8s.GetAllPodsList, dir, errLog)
-	getListAndEncodeToYAML("services", c.sbm.k8s.GetAllServicesList, dir, errLog)
-	getListAndEncodeToYAML("deployments", c.sbm.k8s.GetAllDeploymentsList, dir, errLog)
-	getListAndEncodeToYAML("daemonsets", c.sbm.k8s.GetAllDaemonSetsList, dir, errLog)
-	getListAndEncodeToYAML("statefulsets", c.sbm.k8s.GetAllStatefulSetsList, dir, errLog)
-	getListAndEncodeToYAML("jobs", c.sbm.k8s.GetAllJobsList, dir, errLog)
-	getListAndEncodeToYAML("cronjobs", c.sbm.k8s.GetAllCronJobsList, dir, errLog)
-	getListAndEncodeToYAML("nodes", c.sbm.k8s.GetAllNodesList, dir, errLog)
-	getListAndEncodeToYAML("configmaps", c.sbm.k8s.GetAllConfigMaps, dir, errLog)
-	getListAndEncodeToYAML("volumeattachments", c.sbm.k8s.GetAllVolumeAttachments, dir, errLog)
-	getListAndEncodeToYAML("nodemetrics", c.sbm.k8sMetrics.GetAllNodeMetrics, dir, errLog)
-	getListAndEncodeToYAML("podmetrics", c.sbm.k8sMetrics.GetAllPodMetrics, dir, errLog)
+func (c *Cluster) generateKubernetesClusterYAMLs(dir string, errLog io.Writer) {
+	toDir := filepath.Join(dir, "kubernetes")
+	getListAndEncodeToYAML("nodes", c.sbm.k8s.GetAllNodesList, toDir, errLog)
+	getListAndEncodeToYAML("volumeattachments", c.sbm.k8s.GetAllVolumeAttachments, toDir, errLog)
+	getListAndEncodeToYAML("nodemetrics", c.sbm.k8sMetrics.GetAllNodeMetrics, toDir, errLog)
 }
 
-func (c *Cluster) generateSupportBundleYAMLsForHarvester(dir string, errLog io.Writer) {
+func (c *Cluster) generateKubernetesNamespacedYAMLs(namespace string, dir string, errLog io.Writer) {
+	toDir := filepath.Join(dir, "kubernetes")
+	getListAndEncodeToYAML("events", wrap(namespace, c.sbm.k8s.GetAllEventsList), toDir, errLog)
+	getListAndEncodeToYAML("pods", wrap(namespace, c.sbm.k8s.GetAllPodsList), toDir, errLog)
+	getListAndEncodeToYAML("services", wrap(namespace, c.sbm.k8s.GetAllServicesList), toDir, errLog)
+	getListAndEncodeToYAML("deployments", wrap(namespace, c.sbm.k8s.GetAllDeploymentsList), toDir, errLog)
+	getListAndEncodeToYAML("daemonsets", wrap(namespace, c.sbm.k8s.GetAllDaemonSetsList), toDir, errLog)
+	getListAndEncodeToYAML("statefulsets", wrap(namespace, c.sbm.k8s.GetAllStatefulSetsList), toDir, errLog)
+	getListAndEncodeToYAML("jobs", wrap(namespace, c.sbm.k8s.GetAllJobsList), toDir, errLog)
+	getListAndEncodeToYAML("cronjobs", wrap(namespace, c.sbm.k8s.GetAllCronJobsList), toDir, errLog)
+	getListAndEncodeToYAML("configmaps", wrap(namespace, c.sbm.k8s.GetAllConfigMaps), toDir, errLog)
+	getListAndEncodeToYAML("podmetrics", wrap(namespace, c.sbm.k8sMetrics.GetAllPodMetrics), toDir, errLog)
```

Comment:
prefer to call with `GetAllNamespacedEvents` as getAll sounds a none-namespace scope to me.

Reply:
I'm not sure if that's necessary because the parameter is already a namespace.